generator/src/test/java/com/scottlogic/deg/generator/cucumber/InSet.feature

```diff
@@ -3,988 +3,1651 @@ Feature: User can specify that a field value belongs to a set of predetermined o
+  Then the following data should be generated:
+      | foo        |
+      | null       |
+      | <setValue> |
+  Examples:
+      | setValue                | type       |
+      | 1                       | "string"   |
+      | 2019-01-01T00:00:00.000 | "string"   |
+      | "a"                     | "numeric"  |
+      | 2019-01-01T00:00:00.000 | "numeric"  |
+      | 1                       | "temporal" |
+      | "a"                     | "temporal" |
+
+Scenario Outline: Not 'inSet' type values with non-contradicting 'ofType' are successful
+  Given there is a field foo
+    And foo is anything but in set:
+      | <setValue> |
+    And foo is of type <type>
+    And foo is in set:
+      | 1                       |
+      | "a"                     |
+      | 2019-01-01T00:00:00.000 |
+  Then the following data should be generated:
+      | foo             |
+      | null            |
+      | <expectedValue> |
+  Examples:
+      | setValue                | type       | expectedValue           |
+      | 1                       | "string"   | "a"                     |
+      | 2019-01-01T00:00:00.000 | "string"   | "a"                     |
+      | "a"                     | "numeric"  | 1                       |
+      | 2019-01-01T00:00:00.000 | "numeric"  | 1                       |
+      | 1                       | "temporal" | 2019-01-01T00:00:00.000 |
+      | "a"                     | "temporal" | 2019-01-01T00:00:00.000 |
+
+
+Scenario Outline: Running a 'inSet' of string values with a contradicting ofType emits null
+  Given there is a field foo
+    And foo is in set:
+      | "1" |
+      | "2" |
+      | "3" |
+    And foo is of type <type>
+  Then the following data should be generated:
+      | foo  |
+      | null |
+  Examples:
+      | type       |
+      | "numeric"  |
+      | "temporal" |
+
+Scenario Outline: Running a 'inSet' of numeric values with a contradicting ofType emits null
+  Given there is a field foo
+    And foo is in set:
+      | 1 |
+    And foo is of type <type>
+  Then the following data should be generated:
+      | foo  |
+      | null |
+  Examples:
+      | type       |
+      | "string"   |
+      | "temporal" |
+
+Scenario Outline: Running a 'inSet' of date values with a contradicting ofType emits null
+  Given there is a field foo
+    And foo is in set:
+      | 2010-01-01T00:00:00.000 |
+      | 2010-01-01T00:00:00.001 |
+      | 2011-01-01T00:00:00.000 |
+    And foo is of type <type>
+  Then the following data should be generated:
+      | foo  |
+      | null |
+  Examples:
+      | type      |
+      | "numeric" |
+      | "string"  |
+
+Scenario Outline: : 'InSet' equal to a type value and with not 'ofType' for the same type should emit null
+  Given there is a field foo
+    And foo is in set:
+      | <typeValue> |
+    And foo is anything but of type <type>
+  Then the following data should be generated:
+      | foo  |
+      | null |
+  Examples:
```

Comment:
What does this 
Examples :
thing mean?

Reply:
See the cucumber spec: https://docs.cucumber.io/gherkin/reference/#scenario-outline